src/ucp/tag/rndv.c

```diff
@@ -30,7 +31,9 @@ static int ucp_rndv_is_recv_pipeline_needed(ucp_request_t *rndv_req,
     ucs_for_each_bit(md_index,
                      ucp_ep_config(rndv_req->send.ep)->key.rma_bw_md_map) {
         md_attr = &rndv_req->send.ep->worker->context->tl_mds[md_index].attr;
-        if (md_attr->cap.reg_mem_types & UCS_BIT(mem_type)) {
+        if (md_attr->cap.reg_mem_types & UCS_BIT(mem_type) &&
+            (!remote_address || /* remote is host memory */
```

Comment:
q: could you pls elaborate a little bit more on what these checks do?
I think more documentation would be useful here.

Reply:
wanted to disqualify recv side pipeline H-D inter node(IB) case if GDR is on.
and enable recv GET pipeline H->D intra node(CUDA_IPC) case because CUDA_IPC works only between Device  buffers

Reply:
@bureddy thank you!